Backend/src/main/java/com/oops/backend/Backend/controllers/BackendController.java

```diff
@@ -0,0 +1,119 @@
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestPart;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.multipart.MultipartFile;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.oops.backend.Backend.models.Solver;
+import com.oops.backend.Backend.requests.CancelSolverRequest;
+import com.oops.backend.Backend.requests.CancelTaskRequest;
+import com.oops.backend.Backend.requests.CancelUserTasksRequest;
+import com.oops.backend.Backend.requests.SolveRequest;
+import com.oops.backend.Backend.services.BackendService;
+
+@RestController
+public class BackendController {
+
+    public BackendController() {
+    }
+
+    private BackendService backendService = new BackendService();
+
+    @PostMapping("/Solve")
+    public ResponseEntity<String> startSolvers(@RequestBody SolveRequest request) {
+        backendService.postSolversToSolverManager(request);
+
+        String taskID = request.getProblemID();
+        return new ResponseEntity<String>(taskID, HttpStatus.OK);
+    }
+
+    @GetMapping("/ProblemInstance/{ProblemID}")
+    public ResponseEntity<String> getMZNData(@PathVariable String ProblemID) {
+        String mznData = backendService.getMznData(ProblemID);
+        return new ResponseEntity<String>(mznData, HttpStatus.OK);
+    }
+
+    @PostMapping("/ProblemInstance")
+    public ResponseEntity<String> addMZNToStorage(
+            @RequestPart(name = "mznFile", required = false) MultipartFile mznData,
+            @RequestPart(name = "dznFile", required = false) MultipartFile dznData,
+            @RequestPart String UserID) throws IOException, InterruptedException {
+
+        if (mznData != null && dznData == null) {
+            String problemId = backendService.addMznData(UserID, mznData);
+            ResponseEntity.status(HttpStatus.OK).body(problemId);
+        } else if (mznData == null && dznData != null) {
+            String dataID = backendService.addDznData(UserID, dznData);
+            ResponseEntity.status(HttpStatus.OK).body(dataID);
+        } else if (mznData != null && dznData != null) {
+            String problemId = backendService.addMznData(UserID, mznData);
+            String dataID = backendService.addDznData(UserID, dznData);
+            ResponseEntity.status(HttpStatus.OK).body(problemId);
+        }
+
+        return ResponseEntity.status(HttpStatus.I_AM_A_TEAPOT).body("No file data provided, nothing to upload");
+    }
+
+    @GetMapping("/DataInstance/{ProblemID}")
+    public ResponseEntity<String> getDZNData(@PathVariable String ProblemID) {
+        String dznData = backendService.getDznData(ProblemID);
+        return new ResponseEntity<String>(dznData, HttpStatus.OK);
+    }
+
+    @GetMapping("/Solver")
+    public ResponseEntity<String> getSolvers() throws JsonProcessingException {
```

Comment:
I dont know how spring reacts to endpoints throwing exceptions, I suggest catching the exception in the method, and then returning a response body indicating that something went wrong.